extensions/podman/src/extension.ts

```diff
@@ -884,6 +903,58 @@ export function registerOnboardingMachineExistsCommand(): extensionApi.Disposabl
   });
 }
 
+export function registerOnboardingUnsupportedPodmanMachineCommand(): extensionApi.Disposable {
+  return extensionApi.commands.registerCommand('podman.onboarding.checkUnsupportedPodmanMachine', async () => {
+    let isUnsupported = false;
+    const installedPodman = await getPodmanInstallation();
+    if (installedPodman) {
+      isUnsupported = await shouldNotifyQemuMachinesWithV5(installedPodman);
+    }
+
+    extensionApi.context.setValue('unsupportedPodmanMachine', isUnsupported, 'onboarding');
+  });
+}
+
+export function registerOnboardingRemoveUnsupportedMachinesCommand(): extensionApi.Disposable {
+  return extensionApi.commands.registerCommand('podman.onboarding.removeUnsupportedMachines', async () => {
+    // only on macOS
+    // do not check if version is v5 as it is being checked by the command that triggers this one
+    if (extensionApi.env.isMac) {
+      // remove the qemu machines folder
+      const qemuSharePath = path.resolve(os.homedir(), appHomeDir(), 'machine', 'qemu');
+      const qemuConfigPath = path.resolve(os.homedir(), appConfigDir(), 'machine', 'qemu');
+
+      // remove folders if exists
+      const foldersToRemove = [];
+      if (fs.existsSync(qemuSharePath)) {
+        foldersToRemove.push(qemuSharePath);
+      }
+      if (fs.existsSync(qemuConfigPath)) {
+        foldersToRemove.push(qemuConfigPath);
+      }
+
+      // prompt the user to confirm
+      const result = await extensionApi.window.showWarningMessage(
+        'Confirm removal of the unsupported qemu podman machines',
```

Comment:
We don't say that these are qemu prior to this, so I think we should just leave it out, otherwise it could be confusing or distracting to some users.

Even though they're moving to 5.0 and won't have access to these machines, they could still have work/images in them. Maybe we should be more explicit too? e.g. "Removing old unsupported Podman machines will delete all of their data. Confirm approval?"

Reply:
will change 👍